pkg/webhook/controlplane/ensurer.go

```diff
@@ -79,8 +80,13 @@ func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, gctx gconte
 		return err
 	}
 
+	k8sGreaterEqual127, err := version.CompareVersions(cluster.Shoot.Spec.Kubernetes.Version, ">=", "1.27")
```

Comment:
Please, check my comments in https://github.com/gardener/gardener-extension-provider-aws/pull/759. They apply for this file as well.